nomad/acl_endpoint.go

```diff
@@ -1912,9 +1912,11 @@ func (a *ACL) UpsertAuthMethods(
 			}
 		}
 
-		// if PKCE is not explicitly disabled, enable it.
-		if authMethod.Config.OIDCDisablePKCE == nil {
-			authMethod.Config.OIDCDisablePKCE = pointer.Of(false)
+		// if this is a new auth method, and PKCE is not explicitly disabled
+		// (by setting Enable=false), then enable it. existing auth methods
+		// need it to be enabled explicitly (Enable=True).
+		if existingMethod == nil && authMethod.Config.OIDCEnablePKCE == nil {
```

Comment:
I just noticed that the canonicalize calls in `nomad/structs` don't force `authMethod.Config` to be non-nil. Are we catching that somewhere else?

Reply:
:+1: Thanks!